platform-sdk/swirlds-common/src/main/java/com/swirlds/common/wiring/model/DeterministicWiringModel.java

```diff
@@ -0,0 +1,171 @@
+/*
+ * Copyright (C) 2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.common.wiring.model;
```

Comment:
Should live in a test package not visible to production since the intent is test use.

Reply:
In principal I agree with the idea that test code belongs in the test framework directory, but in this particular case there are a lot of hoops we'd have to jump through in order to move this. The primary problem is that there are a lot of package private things in the `internal` package that the deterministic scheduler builder uses in order to do what it needs to do. If I were to extract that code then I'd lose access to these package private classes and methods. In its current form, it's not possible to implement a deterministic task scheduler using only the public API in the wiring framework.